driver/native.go

```diff
@@ -1,14 +1,74 @@
 package driver
 
-// AndroidContext is passed to the `RunNative` callback when it is executed on an Android device.
-// The VM, Env and Ctx pointers are reqiured to make various calls into JVM methods.
+// NativeWindow is an extension interface for `fyne.Window` that gives access
+// to platform-native features of application windows.
+//
+// Since: 2.5
+type NativeWindow interface {
+	// RunNative  provides a way to execute code within the platform-specific runtime context for a window.
+	// The context types are defined in the `driver` package and the specific context passed will differ by platform.
+	RunNative(func(context any) error) error
+}
+
+// AndroidContext is passed to the RunNative callback when it is executed on an Android device.
+// The VM, Env and Ctx pointers are required to make various calls into JVM methods.
 //
 // Since: 2.3
 type AndroidContext struct {
 	VM, Env, Ctx uintptr
 }
 
-// UnknownContext is passed to the `RunNative` callback when it is executed on devices without special native context.
+// AndroidWindowContext is passed to the NativeWindow.RunNative callback when it is executed
+// on an Android device. The NativeWindow field is of type `*C.ANativeWindow`.
+// The VM, Env and Ctx pointers are required to make various calls into JVM methods.
+//
+// Since: 2.5
+type AndroidWindowContext struct {
+	AndroidContext
+	NativeWindow uintptr
+}
+
+// UnknownContext is passed to the RunNative callback when it is executed
+// on devices or windows without special native context.
 //
 // Since: 2.3
 type UnknownContext struct{}
+
+// WindowsWindowContext is passed to the NativeWindow.RunNative callback
+// when it is executed on a Microsoft Windows device.
+//
+// Since: 2.5
+type WindowsWindowContext struct {
+	// HWND is the window handle for the native window.
+	HWND uintptr
+}
+
+// MacWindowContext is passed to the NativeWindow.RunNative callback
+// when it is executed on a macOS device.
+//
+// Since: 2.5
+type MacWindowContext struct {
+	// NSWindow is the window handle for the native window.
+	NSWindow uintptr
+}
+
+// X11WindowContext is passed to the NativeWindow.RunNative callback
+// when it is executed on a device with the X11 windowing system.
+//
+// Since: 2.5
+type X11WindowContext struct {
+	// WindowHandle is the window handle for the native X11 window.
+	WindowHandle string
+}
+
+// WaylandWindowContext is passed to the NativeWindow.RunNative callback
+// when it is executed on a device with the Wayland windowing system.
+//
+// Since: 2.5
+type WaylandWindowContext struct {
+	// WaylandSurface is the handle to the native Wayland surface.
+	WaylandSurface uintptr
+
+	// EGLSurface is the handle to the native EGL surface.
+	EGLSurface uintptr
+}
```

Comment:
Would it be or be more helpful to give access to the underlying `*glfw.Window` type with the desktop driver? It makes the interfaces much more useful but also a bit harder to use (which I think is fine if someone wants to use this in the first place).

Reply:
I wanted to keep GLFW as an implementation detail - since theoretically we could move to a different windowing manager (or do it all native), even though we currently have no plans to. Thoughts?

Reply:
Plus we can always add it to the window context structs later if needed, since it would be a purely additive change

Reply:
> I wanted to keep GLFW as an implementation detail and not expose it in the public API 

Agreed - there is no guarantee that GLFW remains the native implementation. Pointers to the native system are at least specific to the platform the developer is looking to get access to.

Reply:
Hmm, yeah. I do agree with you there to be honest. My idea was to allow access to the glfw window so we could have a nice way of closing out issues like https://github.com/fyne-io/fyne/issues/1129 and https://github.com/fyne-io/fyne/issues/1155 by saying that it is now doable but not something that we officially support through our own APIs.

Reply:
Those should still be do-able by going through the platform-native C APIs I'd guess.